shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1062,10 +1212,31 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
     return _cachedFirstRect;
   }
 
-  [_textInputDelegate showAutocorrectionPromptRectForStart:start
-                                                       end:end
-                                                withClient:_textInputClient];
-  // TODO(cbracken) Implement.
+  if (!_scribbleInProgress && !_scribbleFocusing && !_scribbleFocused) {
+    [_textInputDelegate showAutocorrectionPromptRectForStart:start
+                                                         end:end
+                                                  withClient:_textInputClient];
+  }
+
+  NSUInteger first = start;
+  if (end < start) {
+    first = end;
+  }
+  FlutterTextRange* textRange = [[FlutterTextRange
+      rangeWithNSRange:fml::RangeForCharactersInRange(self.text, NSMakeRange(0, self.text.length))]
+      copy];
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    if ([_selectionRects[i][4] unsignedIntegerValue] <= first &&
+        ((i + 1 == [_selectionRects count] && textRange.range.length > first) ||
+         (i + 1 < [_selectionRects count] &&
+          [_selectionRects[i + 1][4] unsignedIntegerValue] > first))) {
```

Comment:
I recommend pulling this bool expression into a function or a local variable so people don't have to parse what it is trying to say.